### PR TITLE
Modify the processer to support the deletion of the pkl Reco files after skims production

### PR DESCRIPTION
**Pull request by DelloStritto**

### CODE REVIEW

**Comment by qgp** [comment]
Just to clarify, the intention is to support the deletion of the reco pkls, not to delete them automatically in the processer, right?

**Comment by DelloStritto** [comment]
> Just to clarify, the intention is to support the deletion of the reco pkls, not to delete them automatically in the processer, right?

Yes, so that if you need to free up some space you can delete them without preventing the execution of all the subsequent steps.

File: machine_learning_hep/processer.py [hunk 1]
```diff
@@ -181,9 +182,14 @@ def __init__(self, case, datap, run_param, mcordata, p_maxfiles, # pylint: disab
         if os.path.isdir(self.d_root):
             self.l_path = list_folders(self.d_root, self.n_root, self.p_maxfiles,
                                        self.select_jobs)
-        else:
+        elif is_file_indir(self.d_pkl, self.n_reco):
```
Comment (qgp) [inline machine_learning_hep/processer.py]: Why not simply this?
```suggestion
        elif glob.glob(f"{self.d_pkl}/**/{self.n_reco}", recursive=True):
```

File: machine_learning_hep/utilities.py [hunk 2]
```diff
@@ -194,6 +194,15 @@ def list_folders(main_dir, filenameinput, maxfiles, select=None): # pylint: disa
...
 
+def is_file_indir(main_dir, filenameinput):
+    """
+    Check if a file is contained in a directory or all its subdirectories
+    """
+    for _, _, files in os.walk(main_dir):
+        if filenameinput in files:
+            return True
+    return False
+
```
Comment (qgp) [inline machine_learning_hep/utilities.py]: This look like an unnecessary implementation of functionality provided by glob (see above).

**Review by qgp — CHANGES_REQUESTED** [review]

**Pull request merged**